backend.native/tests/interop/platform_zlib.kt

```diff
@@ -21,7 +21,7 @@ fun main(args: Array<String>) = memScoped {
         }.ptr
 
         if (inflateInit2(z, -15) == Z_OK && inflate(z, Z_FINISH) == Z_STREAM_END && inflateEnd(z) == Z_OK)
-            println(buffer.stringFromUtf8())
+            println(buffer.decodeZeroTerminatedToString())
```

Comment:
`decodeZeroTerminatedToString` looks somewhat ugly. Nobody care if it is decoding or copying, just `fromTerminatedUtf8` looks sufficient.

Reply:
We want this name to follow the convention _"somethingToString"_. 
- "ToString" shows the target type one gets in the result
- "something" can be either a verb, like `decode`, or a description of source content, like `hex`,  `utf8` or `terminatedUtf8`

So another alternative could be `terminatedUtf8ToString()`. Though this would be at odds with our original goal not to have "utf8" in the name of `decodeToString` to allow providing an overload with the same name in kotlinx-io taking another encoding as an argument. 

Can we count on never needing encoding other than utf-8 in this use case?

Reply:
We must have similar convertors from UTF-16 and UTF-32. https://github.com/JetBrains/kotlin-native/blob/bff177a0728ac26f0839f9e52dddd49c88a18591/Interop/Runtime/src/main/kotlin/kotlinx/cinterop/Utils.kt#L509 have those extensions for C pointers. Maybe toKStringFromUtf[8|16|32] should be OK.

Reply:
We discussed and liked this idea.

Given that there's `CPointer<ByteVar>.toKString()` in `kotlinx.cinterop`, we can provide the overload `ByteArray.toKString()` beside.

Reply:
done